kolibri/core/errorreports/middleware.py

```diff
@@ -0,0 +1,37 @@
+import logging
+import traceback
+
+from django.db import IntegrityError
+
+from .models import ErrorReports
+
+
+class ErrorReportingMiddleware:
+    """
+    Middleware to log exceptions to the database.
+    ref: https://docs.djangoproject.com/en/5.0/topics/http/middleware/#writing-your-own-middleware
+    """
+
+    def __init__(self, get_response):
+        self.get_response = get_response
+        self.logger = logging.getLogger(__name__)
+
+    def __call__(self, request):
+        response = self.get_response(request)
+        return response
+
+    def process_exception(self, request, exception):
+        error_message = str(exception)
+        traceback_info = traceback.format_exc()
+        self.logger.error("Unexpected Error: %s", error_message)
+
+        try:
+            self.logger.error("Saving error report to the database.")
+            ErrorReports.insert_or_update_error(
+                "Backend", error_message, traceback_info
```

Comment:
I think we should define this as a constant and reference it here? See [here](https://github.com/learningequality/kolibri/blob/distributed-error-reporting/kolibri/core/errorreports/models.py#L53-L56) in your previous pr.

In our `core/errorreports`, we can create a `constants` folder and add an `error_kinds.py`.

Reply:
Certainly!